cmd/lk/join.go

```diff
@@ -368,3 +371,176 @@ func publishReader(room *lksdk.Room,
+	quality livekit.VideoQuality
+	name    string
+}
+
+// handleSimulcastPublish handles publishing multiple H.264 streams as a simulcast track
+func handleSimulcastPublish(room *lksdk.Room, urls []string, fps float64, onPublishComplete func(*lksdk.LocalTrackPublication)) error {
+	// Parse all URLs
+	var layers []simulcastLayer
+	for _, url := range urls {
+		parts, err := parseSimulcastURL(url)
+		if err != nil {
+			return fmt.Errorf("invalid simulcast URL %s: %w", url, err)
+		}
+		if parts != nil {
+			layers = append(layers, simulcastLayer{
+				url:   url,
+				parts: parts,
+			})
+		}
+	}
+
+	if len(layers) == 0 {
+		return fmt.Errorf("no valid simulcast URLs provided")
+	}
+
+	// Sort streams by width to determine quality levels
+	sort.Slice(layers, func(i, j int) bool {
+		return layers[i].parts.width < layers[j].parts.width
+	})
+
+	// Assign quality levels based on stream count and order
+	if len(layers) == 2 {
+		// 2 streams: low and high quality
+		layers[0].quality = livekit.VideoQuality_LOW
+		layers[0].name = "low"
+		layers[1].quality = livekit.VideoQuality_HIGH
+		layers[1].name = "high"
+	} else if len(layers) == 3 {
+		// 3 streams: low, medium, high quality
+		layers[0].quality = livekit.VideoQuality_LOW
+		layers[0].name = "low"
+		layers[1].quality = livekit.VideoQuality_MEDIUM
+		layers[1].name = "medium"
+		layers[2].quality = livekit.VideoQuality_HIGH
+		layers[2].name = "high"
+	} else {
+		return fmt.Errorf("simulcast requires 2 or 3 streams, got %d", len(layers))
+	}
+
+	// Create tracks for each stream
+	var tracks []*lksdk.LocalTrack
+	var trackNames []string
+
+	// Track completion - if any stream ends, signal completion
+	var pub *lksdk.LocalTrackPublication
+	completionSignaled := false
+	var completionMutex sync.Mutex
+
+	signalCompletion := func() {
+		completionMutex.Lock()
+		defer completionMutex.Unlock()
+		if !completionSignaled && onPublishComplete != nil {
+			completionSignaled = true
+			onPublishComplete(pub)
+		}
+	}
+
+	for _, layer := range layers {
+		track, err := createSimulcastVideoTrack(layer.parts, layer.quality, fps, signalCompletion)
+		if err != nil {
+			// Clean up any tracks we've already created
+			for _, t := range tracks {
+				t.Close()
+			}
+			return fmt.Errorf("failed to create %s quality track (%dx%d): %w",
+				layer.name, layer.parts.width, layer.parts.height, err)
+		}
+		tracks = append(tracks, track)
+		trackNames = append(trackNames, fmt.Sprintf("%s(%dx%d)", layer.name, layer.parts.width, layer.parts.height))
+	}
+
+	// Publish simulcast track
+	var err error
+	pub, err = room.LocalParticipant.PublishSimulcastTrack(tracks, &lksdk.TrackPublicationOptions{
+		Name: "simulcast",
+	})
+	if err != nil {
+		// Clean up tracks on publish failure
+		for _, track := range tracks {
+			track.Close()
+		}
+		return fmt.Errorf("failed to publish simulcast track: %w", err)
+	}
+
+	fmt.Printf("Successfully published H.264 simulcast track with qualities: %v\n", trackNames)
```

Comment:
It's not necessary, but a table of the published tracks and their properties might be nice here.

Reply:
Yeah, we use it in tons of places in the CLI, for example when you run `lk project list`: https://github.com/livekit/livekit-cli/blob/b01c31b12bc6163528bd67298d97f15b5101c849/cmd/lk/project.go#L254-L275